seclea_ai/internal/file_processor.py

```diff
@@ -22,15 +24,82 @@ class FileProcessor:
     Something to wrap backend requests. Maybe use to change the base url??
     """
 
-    def __init__(self, project_name, organization, transmission, api):
+    def __init__(self, project_name, organization, transmission, api, auth_service):
         # setup some defaults
         self._transmission = transmission
         self._project = None
         self._project_name = project_name
         self._organization = organization
         self._cache_dir = os.path.join(tempfile.gettempdir(), ".seclea/cache")
         self._api = api
+        self._auth_service = auth_service
         self._dbms = MyDatabase()
+        self._sender_q = Queue()
+
+    def writer(self, _storage_q: Queue):
+        self._storage_q = _storage_q
+        while not self._storage_q.empty():
+            obj = self._storage_q.get()
+
+            if obj:
+                if obj["function"] == "upload_dataset":
+                    self._save_dataset(
+                        obj["project"],
+                        obj["dataset"],
+                        obj["dataset_name"],
+                        obj["metadata"],
+                        obj["parent_hash"],
+                        obj["transformation"],
+                    )
+                if obj["function"] == "upload_training_run":
+                    self.upload_training_run(
+                        model=obj["model"],
+                        train_dataset=obj["train_dataset"],
+                        test_dataset=obj["test_dataset"],
+                        val_dataset=obj["val_dataset"],
+                        project=obj["project"],
+                    )
+                if obj["function"] == "_save_model_state":
+                    self._save_model_state(
+                        model=obj["model"],
+                        training_run_pk=obj["training_run_pk"],
+                        sequence_num=obj["sequence_num"],
+                        final=obj["final"],
+                        model_manager=obj["model_manager"],
+                    )
+
+    def sender(self):
+        while not self._sender_q.empty():
```

Comment:
Not sure if this exits when the queue is empty or just pauses - will make a big difference.

Reply:
Actually i have checked when queue has no data then it gives response but will try to do it different way

Reply:
I have refactored, resolving this.